feat(get-token): update coin image

Matinj031 opened the pull request.

skajosborn committed:
feat(get-token): update coin image size, tilt animation, and font styles

skajosborn committed:
fix(get-token): refine layout and styling for about, earn, and trust sections

skajosborn committed:
feat(get-token): made several updates to  page to improve layout, responsiveness and overall appearance

skajosborn committed:
Fix build error in  page

skajosborn committed:
chore(get-token): replaced PNGs with SVGs for better clarity and scalability

skajosborn committed:
Refactor: Made several updates and improvements to  token pg to improve overall appearance and layout including svgs in place of pngs

skajosborn committed:
Refactor: restored original icons in community section

skajosborn committed:
fixed dot on live trade badge and darkened icon backgrounds in trust section

skajosborn committed:
Enhanced tokenomics and how to buy pages and improved styling

skajosborn committed:
Update to Tokenomics and How to Buy sections of  pg. Added popups, improved layout and design.

skajosborn committed:
Fixed mobile responsiveness to Tokenomics section

skajosborn committed:
feat: Improve layout and responsiveness for Buy, Tokenomics, and Hero sections

skajosborn committed:
fixed casing error in TokenPurchse import

Matinj031 committed:
Merge branch 'upgrade-v3' of https://github.com/GamaEdtech/front into skajosborn/upgrade-v3

Matinj031 committed:
feat(token image): Updating the coin image to match the Figma design, fix husky linter errors

Matinj031 committed:
feat(tokenomics): Improves tokenomics modal responsiveness

Matinj031 committed:
feat(statsSection): Improves token stats section responsiveness

Enhances the responsiveness of the token stats section by introducing a mobile-friendly layout.

This change restructures the component to display price, token list, and chart in a vertical stack on smaller screens, improving usability on mobile devices.

It also corrects a typo in "Circulating Supply".

Matinj031 committed:
feat(tokenomics image): Improves tokenomics image appearance
Applies a multiply blend mode to the

Matinj031 committed:
feat: add file path case sensitivity checker script

- Add scripts/check-file-paths.js to detect case sensitivity issues
- Helps prevent deployment issues on Linux/macOS systems
- Focuses on image files in /images/ directory
- Provides clear error reporting and suggestions
- Add npm script 'check-paths' to run the validation
- Install glob dependency for file pattern matching

Usage: npm run check-paths

Matinj031 committed:
feat: add useSafeImage composable for robust image handling

- Add composables/useSafeImage.js for safe image path management
- Provides validateImagePath() for checking image existence
- Includes getSafeImagePath() for path normalization
- Offers createImageLoadHandler() for better error handling
- Helps developers catch case sensitivity issues during development
- Includes console warnings for debugging image loading problems

Usage: const { createImageLoadHandler } = useSafeImage()

Matinj031 committed:
feat: add development plugin for runtime image validation

- Add plugins/file-path-validator.client.js for development mode
- Automatically validates image src attributes in browser
- Provides console warnings for failed image loads
- Runs validation on initial load and route changes
- Only active in development mode for performance
- Helps catch case sensitivity issues during development

The plugin automatically runs and shows warnings in browser console

Matinj031 committed:
fix: resolve case sensitivity issue in TokenPurchase component

- Fix CTA_People_mobile.svg to CTA_People_Mobile.svg (case mismatch)
- Remove duplicate slash in image path (/images/token//CTA/ -> /images/token/CTA/)
- This resolves deployment issues on case-sensitive file systems (Linux/macOS)
- Prevents 404 errors when deployed to Linux servers

The file exists as CTA_People_Mobile.svg but was referenced as CTA_People_mobile.svg

Matinj031 committed:
feat: enhance StatsSection with mobile optimization and safe image handling

- Implement responsive layout with vertical stacking on mobile
- Increase mobile font sizes (price: 10px  16px, tokens: 10px  12px)
- Enhance mobile chart height from 145px to 250px for better visibility
- Add proper touch targets (44px minimum) for accessibility
- Integrate useSafeImage composable for better error handling
- Fix typo: 'Circuleating'  'Circulating' Supply
- Add lazy loading and error handling for token icons
- Improve mobile user experience with better spacing and typography

Mobile users now have significantly improved readability and usability

Matinj031 committed:
feat: enhance ESLint config with file path validation setup

- Add file-path-case-sensitivity configuration section
- Prepare ESLint for future file path validation rules
- Document the approach for handling case sensitivity in linting
- This complements the standalone check-paths script

The actual validation is handled by the separate check-paths script for better performance

Matinj031 committed:
feat: integrate case sensitivity check into pre-commit hook

- Add check-paths script to pre-commit workflow
- Automatically validate file paths before each commit
- Focus on critical case sensitivity issues (not missing files)
- Prevent case sensitivity issues from being committed
- Works on both Windows and Unix-like systems
- Provides immediate feedback to developers

Now every commit will be validated for case sensitivity issues automatically.
Missing files are reported but don't block commits.

Matinj031 committed:
docs: add comprehensive documentation for file path validation system

- Add FILE_PATH_VALIDATION.md with complete usage guide
- Document all implemented tools and their purposes
- Include best practices for file naming and development workflow
- Provide troubleshooting guide for common issues
- Add examples of correct vs incorrect usage
- Written in Persian for better team understanding

This documentation helps team members understand and use the new validation system effectively.

sanaderi committed:
Delete docs/FILE_PATH_VALIDATION.md

sanaderi commented on scripts/check-file-paths.js:
```diff
@@ -0,0 +1,186 @@
+#!/usr/bin/env node
+
```
Keep scripts like check-file-paths.js in a separate directory (e.g., scripts/) that is NOT publicly served.

sanaderi commented on docs/FILE_PATH_VALIDATION.md:
Change this file to english. never used any local languge in project

Matinj031 committed:
Merge branch 'upgrade-v3' of https://github.com/GamaEdtech/front into skajosborn/upgrade-v3

Matinj031 committed:
docs: translate FILE_PATH_VALIDATION.md to English

- Convert all Persian content to English as per project standards
- Maintain all technical information and examples
- Keep the same structure and formatting
- Ensure consistency with project's English-only documentation policy

This addresses the code review feedback about using local language in project files.

Matinj031 committed:
docs: clarify that scripts are development-only and not publicly served

- Add clear documentation that scripts/check-file-paths.js is for development use
- Confirm that scripts directory is not publicly accessible
- Address security concern about keeping development scripts separate
- Scripts directory is properly isolated from public serving by Nuxt.js default behavior
- Fix ESLint trailing spaces issue

This addresses the code review feedback about script location and accessibility.

Matinj031 committed:
resolve: keep FILE_PATH_VALIDATION.md documentation

- Resolve merge conflict by keeping the documentation file
- The file was deleted in remote but modified locally
- Documentation is valuable for team understanding of case sensitivity tools
- File has been translated to English as per project standards
